data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -266,13 +291,23 @@ private CloseableIterable<T> createDeleteIterable(
         : Deletes.filterDeleted(records, isDeleted, counter);
   }
 
+  static CloseableIterable<Record> openPosDeletes(FileIO io, DeleteFile file) {
+    InputFile input = io.newInputFile(file.path().toString());
```

Comment:
This should use `getInputFile` instead of calling `io.newInputFile`. In Spark and Flink, the input files are already created in a bulk operation.

Reply:
fixed